packages/editor-ui/src/modules/ui.ts

```diff
@@ -124,7 +125,10 @@ const module: Module<IUiState, IRootState> = {
 			focusedMappableInput: '',
 			mappingTelemetry: {},
 		},
+		showNodeCreatorTabs: true,
+		showCreatorPanelScrim: false,
 		mainPanelPosition: 0.5,
+		selectedNodeCreatorType: ALL_NODE_FILTER,
```

Comment:
They don't, but it's less verbose than passing them through via props.
1. `showNodeCreatorTabs` controls rendering of tabs inside MainPanel. If we'd store this state locally just inside `MainPanel`, we'd need to listen for an emitted event both from `TriggerHelperPanel` and `CategorizedItems`. And since `TriggerHelperPanel` is also using `CategorizedItems`, we'd also have to set up a listener there.
2. `showCreatorPanelScrim` in the upcoming PR, we'll show scrim only if the trigger tab is opened via the "Choose a trigger" node, so it'll be handy to only call if then.
3. `selectedNodeCreatorType` is again used in `CategorizedItems`, which are used both in `MainPanel` and `TriggerHelperPanel`, so we'd have to pass it through(MainPanel > CategorizedItems && MainPanel > TriggerHelperPanel > CategorizedItems)
I can create a standalone module for these node creator store props if you think it makes more sense

Reply:
still not convinced like we have discussed... but it's your call... though `showNodeCreatorTabs` seems like it should be a derived property no?